### PR TITLE
28 logout link

### PR DESCRIPTION
**Pull request by benjaminsinaiko**
#28 Logout Link

- Changed default export User action to named exports for User actions. Created resetUserAction to clear user object/return to initial user state of null on logout.

- Create Logout Button and component to clear User store data and User session.

### CODE REVIEW

**Comment by donthedeveloper** [comment]
Hey @benjaminsinaiko, I won't get to this this weekend, but it looks like there are some merge conflicts. Would you mind merging in the most recent `master` and fixing these?

**Comment by donthedeveloper** [comment]
It looks like there is a merge conflict with this.

File: .eslintrc.json [hunk 1]
```diff
@@ -1,11 +1,6 @@
 {
-  "plugins": [
-    "jest"
-  ],
-  "extends": [
-    "airbnb",
-    "plugin:jest/recommended"
-  ],
+  "plugins": ["jest"],
```
Comment (donthedeveloper) [inline .eslintrc.json]: I feel like these aren't the actual linter rules. I'm wondering if you have different settings enabled for yours? Can we just exclude this file from the history here?

File: browser/src/react-redux/App/User/User.actions.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
-const setUser = (user) => {
-  return {
-    type: 'SET_USER',
-    user
-  }
-}
+export const setUserAction = user => ({
```
Comment (donthedeveloper) [inline browser/src/react-redux/App/User/User.actions.js]: Is there a reason you added "Action" to the variable name? I think we can just keep these as `setUser` and `resetUser`.

File: .eslintrc.json (hunk 1 above)
Comment (donthedeveloper) [reply .eslintrc.json]: Actually, someone mentioned that you might be using something like beautify, as these changes don't look like some of our linter rules. I have a feeling when you merge in the latest master, you're going to get blocked from submitting these changes. Are you using any extra extensions that may be doing this besides the ones that I recommend when we initially installed the linter rules?

File: browser/src/react-redux/App/Logout.React.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
```
Comment (donthedeveloper) [inline browser/src/react-redux/App/Logout.React.js]: Let's stick with our current component folder convention. So `App/Logout.React.js` becomes `App/Logout/Logout.react.js`.

File: browser/src/react-redux/App/Nav.react.js [hunk 4]
```diff
@@ -1,22 +1,24 @@
...
+        <Link to="/">Home</Link>
+      </li>
+      <li>
+        <Link to="/signup">Sign Up</Link>
+      </li>
+      <li>
+        <Link to="/login">Login</Link>
+      </li>
+      <li>
+        <Logout />
```
Comment (donthedeveloper) [inline browser/src/react-redux/App/Nav.react.js]: Did we talk about conditionally rendering these based on whether the user was logged in or not? I forgot where we left off with that conversation.

File: browser/src/react-redux/App/User/User.actions.js (hunk 2 above)
Comment (benjaminsinaiko) [reply browser/src/react-redux/App/User/User.actions.js]: We were naming the method in Signup and Login setUser as well, so wasn't liking same name.

File: browser/src/react-redux/App/Logout.React.js (hunk 3 above)
Comment (benjaminsinaiko) [reply browser/src/react-redux/App/Logout.React.js]: Done.

File: browser/src/react-redux/App/Nav.react.js (hunk 4 above)
Comment (benjaminsinaiko) [reply browser/src/react-redux/App/Nav.react.js]: Yeah, that is the plan.  We can talk about how to go about that next week and create a new issue.

File: .eslintrc.json (hunk 1 above)
Comment (benjaminsinaiko) [reply .eslintrc.json]: I am using Prettier.  Isn't this the purpose of the ESLint file, to give rules for all these formatting issues?

File: browser/src/react-redux/App/User/User.actions.js (hunk 2 above)
Comment (donthedeveloper) [reply browser/src/react-redux/App/User/User.actions.js]: What wasn't like the same name? You? Prettier?

File: .eslintrc.json (hunk 1 above)
Comment (donthedeveloper) [reply .eslintrc.json]: I think, as a team, we should be consistently using the same styling/linting tools. Neither of us (I believe Jack doesn't) use prettier. That is something separate from the eslinter tools we have set up. Ideally, we want to get rid of all of these prettier changes that are overcomplicating this PR.

Here are some next steps:
1) Undo the linter changes in files that are _just_ in the diff, as I think most of this is due to prettier.
2) Merge in the latest master.
3) Checkout the README file. I believe I added steps to make sure your local env is setup to follow the ESLint rules we have. Let's make sure you have everything set up properly. DM me if you have any questions about the setup.
4) Turn prettier off for this project.

That should get you in a good place to be consistent with the conventions that Jack and I are following. Also, it'll help clear up the diff in this PR.

File: browser/src/react-redux/App/Nav.react.js [hunk 5]
```diff
@@ -1,22 +1,33 @@
...
+          <li>
+            <Link to="/signup">Sign Up</Link>
+          </li>
+          <li>
+            <Link to="/login">Login</Link>
+          </li>
+          {' '}
+        </>
+      ) : (
         <li>
```
Comment (donthedeveloper) [inline browser/src/react-redux/App/Nav.react.js]: Let's go ahead and hide the logged out link when the user is not logged in.

File: browser/src/react-redux/App/Nav.react.js (hunk 4 above)
Comment (benjaminsinaiko) [reply browser/src/react-redux/App/Nav.react.js]: Done.

File: browser/src/react-redux/App/Nav.react.js (hunk 5 above)
Comment (benjaminsinaiko) [reply browser/src/react-redux/App/Nav.react.js]: This was already done.  Were you seeing logout button when not signed in?

File: browser/src/react-redux/App/User/User.actions.js (hunk 2 above)
Comment (benjaminsinaiko) [reply browser/src/react-redux/App/User/User.actions.js]: Put it back to just resetUser and get this error 
`'resetUser' is already declared in the upper scope.eslint(no-shadow)`

File: browser/src/react-redux/App/Nav.react.js (hunk 5 above)
Comment (donthedeveloper) [reply browser/src/react-redux/App/Nav.react.js]: My mistake! This looks good.

**Pull request merged**